src/views/VideoView/VideoView.tsx

```diff
@@ -21,17 +22,28 @@ import { GetVideo, GetVideoVariables } from '@/api/queries/__generated__/GetVide
 import { formatVideoViewsAndDate } from '@/utils/video'
 import { AddVideoView, AddVideoViewVariables } from '@/api/queries/__generated__/AddVideoView'
 import { ChannelLink } from '@/components'
+import { usePersonalData } from '@/hooks'
 
 const VideoView: React.FC<RouteComponentProps> = () => {
   const { id } = useParams()
   const { loading, data, error } = useQuery<GetVideo, GetVideoVariables>(GET_VIDEO, {
     variables: { id },
   })
   const [addVideoView] = useMutation<AddVideoView, AddVideoViewVariables>(ADD_VIDEO_VIEW)
+  const { state, updateWatchedVideos } = usePersonalData()
+
+  const [startTimestamp, setStartTimestamp] = useState<number>()
+  useEffect(() => {
+    if (startTimestamp) {
```

Comment:
We probably should initialize `startTimestamp` with `null` and here check for  `startTimestamp != null`. Like it is right now, if the initial timestamp is 0, it will get updated once anyway